package.json

```diff
@@ -5,46 +5,29 @@
-    "build": "mkdir build; browserify src/index.js -g yo-yoify -o build/app.js -t [ babelify ]; babel --plugins=transform-es2015-template-literals build/app.js --out-file build/app.js",
+    "browser-test-remote-chrome": "env-cmd .env.sauce.local nightwatch --config nightwatch.js --env chrome",
+    "browser-test-remote-firefox": "env-cmd .env.sauce.local nightwatch --config nightwatch.js --env default",
+    "browser-test-remote-ie": "env-cmd .env.sauce.local nightwatch --config nightwatch.js --env ie",
+    "browser-test-remote-safari": "env-cmd .env.sauce.local nightwatch --config nightwatch.js --env safari",
+    "browser-test-remote-parallel": "env-cmd .env.sauce.local nightwatch --config nightwatch.js --env safari,ie,default,chrome",
+    "build": "mkdir -p build; browserify src/index.js -g yo-yoify -t [ babelify --presets [ es2015 ] ] -o build/app.js",
+    "watch": "mkdir -p build; watchify src/index.js -p [ browserify-livereload --host 127.0.0.1 --port 1337 ] -t [ babelify --presets [ es2015 ] ] -o build/app.js -dv",
     "csslint": "csslint --ignore=order-alphabetical --errors='errors,duplicate-properties,empty-rules' --exclude-list='assets/css/font-awesome.min.css' assets/css/",
     "downloadsolc": "rm soljson.js; wget https://ethereum.github.io/solc-bin/soljson.js",
+    "make-mock-compiler": "node ci/makeMockCompiler.js",
     "lint": "standard",
     "serve": "http-server .",
+    "sauce": "env-cmd .env.sauce.local node sc",
     "test": "standard; babel src --out-dir babelify-src; node test/index.js"
   },
   "devDependencies": {
     "async": "^2.1.2",
-    "babel-cli": "^6.16.0",
-    "babel-eslint": "^7.1.1",
-    "babel-plugin-check-es2015-constants": "^6.8.0",
-    "babel-plugin-transform-es2015-arrow-functions": "^6.8.0",
-    "babel-plugin-transform-es2015-block-scoped-functions": "^6.8.0",
-    "babel-plugin-transform-es2015-block-scoping": "^6.18.0",
-    "babel-plugin-transform-es2015-classes": "^6.18.0",
-    "babel-plugin-transform-es2015-computed-properties": "^6.8.0",
-    "babel-plugin-transform-es2015-destructuring": "^6.18.0",
-    "babel-plugin-transform-es2015-duplicate-keys": "^6.8.0",
-    "babel-plugin-transform-es2015-for-of": "^6.18.0",
-    "babel-plugin-transform-es2015-function-name": "^6.9.0",
-    "babel-plugin-transform-es2015-literals": "^6.8.0",
-    "babel-plugin-transform-es2015-object-super": "^6.8.0",
-    "babel-plugin-transform-es2015-parameters": "^6.18.0",
-    "babel-plugin-transform-es2015-shorthand-properties": "^6.18.0",
-    "babel-plugin-transform-es2015-spread": "^6.8.0",
-    "babel-plugin-transform-es2015-sticky-regex": "^6.8.0",
-    "babel-plugin-transform-es2015-template-literals": "^6.8.0",
-    "babel-plugin-transform-es2015-unicode-regex": "^6.11.0",
-    "babel-plugin-transform-regenerator": "^6.16.1",
-    "babel-polyfill": "^6.22.0",
-    "babelify": "^7.3.0",
+    "babel-preset-es2015": "^6.22.0",
```

Comment:
1. It run's a file watcher, that watches an entry file and everything that is `require('...')`ed for changes (e.g. on file save in your editor) ...and then re-runs browserify.
 
2. additionally running browserify with the live-reload feature will on-the-fly run an additional dev server on your local machine and add a little javascript to your build file, that - when loaded in the browser - will estabish a websocket connection to that additional dev server and listen for updates.

3. once the file is re-build with all updates (1.) then the additional server is informed and sends the updates via the web-socket connection to all clients to refresh the browser with the new stuff.


.... there is even more steps - for example browsersync - which will even sync current scroll position and other stuff and if you open the project in multiple browsers and scroll in one -the others will magically scroll too to stay in sync

...a next step is to integrate "hot module reloading" (HMR) which will maintain STATE between refreshes (so if you are in a debugging sessions) and the code updates, then the STATE will be the same, but with the new version of the code - but this takes a little bit MORE effort to integrate :-) ...but might be worth it.

One project that might be worth checking out is https://github.com/amokjs/amok (which plugs into the browser from "behind the scenes" https://www.youtube.com/watch?v=1onqNIPSSeo) but otherwise what will totally do is https://www.npmjs.com/package/browserify-hmr which i have used in the past